plugins/outputs/influxdb/http.go

```diff
@@ -374,6 +374,12 @@ func (c *httpClient) writeBatch(ctx context.Context, db, rp string, metrics []te
 		desc = writeResp.Err
 	}
 
+	//checks for any 4xx code and drops metric and retrying will not make the request work
+	if len(resp.Status) > 0 && resp.Status[0] == '4' && c.config.SkipDatabaseCreation {
+		c.log.Errorf("E! [outputs.influxdb] Failed to write metric (will be dropped: %s): %s\n", resp.Status, desc)
+		return nil
+	}
```

Comment:
I'm wondering if we should handle this case within the database not found error handling which is influxdb.go file from line 222. As the error is the same "database not found", but we weren't handling the case where SkipDatabaseCreation is true.